middleware/validation.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const path = require('path')
+const fs = require('fs')
+
+/*
+  This method creates a joi vaidation schema.
+  It returns null if the schema creation fails and updates the context with an error
+*/
+const createJoiValidationSchema = (ctx) => {
+  let validations
+  const schemaObject = {}
+
+  try {
+    const resourcePath = path.resolve(__dirname, '..', 'endpoints', ctx.resourceName, 'input-validation.json')
+    const file = fs.readFileSync(resourcePath)
+    validations = JSON.parse(file)
+  } catch (error) {
+    ctx.error = error
+    return null
+  }
```

Comment:
Loading these files is an additional responsibility that shouldn't be here. 
What about performing this functionality earlier e.g. routes.js and passing the loaded object around on the ctx?